### PR TITLE
9638 add docs about how to build documentation locally

### LINKED ISSUE #9638: Add documentation on how to build the docs locally
### First check

- [X] I added a descriptive title to this issue.
- [X] I used GitHub search to find a similar request and didn't find it 😇

### Describe the issue

I was looking to make a change to the Prefect docs, and I noticed that the contributing page doesn't mention the best way to build the docs locally.

### Describe the proposed change

Let's add a short section describing:
* What we use to generate our docs (mkdocs)
* How to build the docs locally
* How to make simple updates to the documentation

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by dylanbhughes**
Closes #9638 

### Example
Screenshot 2023-05-18 at 3 19 44 PM

### Checklist

- [X] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [X] This pull request includes tests or only affects documentation.
- [X] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`, `docs`.

### CODE REVIEW

**Review by madkinsz — APPROVED** [review]
Could also add a note about:

- We use mkdocs-material-insiders for production but builds can still happen without the extra plugins
- Internal developers can test insiders by including the `--config-file mkdocs.insiders.yml` flag on `mkdocs` commands
- Deploy previews are available on pull requests

File: docs/contributing/overview.md [hunk 1]
```diff
@@ -194,6 +194,36 @@ prefect dev build-ui
 ```
 </div>
 
+### Docs Development
+
+Prefect users [mkdocs](https://www.mkdocs.org/) for the docs website and the [mkdocs-material](https://squidfunk.github.io/mkdocs-material/) theme.
```
Comment (discdiver) [inline docs/contributing/overview.md:199]:
```suggestion
Prefect uses [mkdocs](https://www.mkdocs.org/) and the [mkdocs-material](https://squidfunk.github.io/mkdocs-material/) theme for the docs website (docs.prefect.io)[https://docs.prefect.io].
```

**Review by discdiver — APPROVED** [review]
Great addition, Dylan! Just caught a typo. Agree with Zanie that it would be nice to add the info suggested in their comment.

File: .gitignore [hunk 2]
```diff
@@ -39,6 +39,7 @@ venv/
 schema.json
 site/
 .cache/
+src/mkdocs-material
```
Comment (madkinsz) [inline .gitignore:42]: 👀 idk about this one haha

**Review by madkinsz — APPROVED** [review]

**Pull request merged**